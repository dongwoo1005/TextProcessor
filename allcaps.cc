#include "allcaps.h"
#include <locale>
using namespace std;

// Refer to allcaps.h for implmentation details

void AllCaps::setSource(istream *in){
	tp.setSource(in);
}

string AllCaps::getWord(){
	string s = tp.getWord();
	int len = s.length();
	for (int i=0; i<len; ++i){
		if (s[i] >= 'a' && s[i] <= 'z'){
			s[i] -= 'a' - 'A';
		}
	}
	return s;
}

bool AllCaps::fail() const {
	return tp.fail();
}
