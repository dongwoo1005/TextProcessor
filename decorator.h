#ifndef __DECORATOR_H__
#define __DECORATOR_H__
#include "textprocess.h"

class Decorator: public TextProcessor{
protected:
	TextProcessor& tp;
	Decorator(TextProcessor& tp): tp(tp){}
	~Decorator(){ delete &tp;}
};

#endif
