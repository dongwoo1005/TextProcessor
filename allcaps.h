#ifndef __ALLCAPS_H__
#define __ALLCAPS_H__
#include "decorator.h"
#include <string>

// ConcreteDecorator AllCaps

class AllCaps: public Decorator {

public:
	AllCaps(TextProcessor& tp): Decorator(tp) {} //ctor

	// void setSource(istream *inp)
	// sets the istream source as inp
	void setSource (std::istream* inp); 

	// string getWord()
	// returns the string in the istream source
	// All letters in the string are presented in uppercase.
	// Other characters remain unchanged
	std::string getWord();

	// bool fail() const
	// returns the failed variable in the class field
	bool fail() const;
};

#endif
