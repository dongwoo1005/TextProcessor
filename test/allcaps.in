stdin allcaps
a1 bc23 abc123
d3son
AllCaps~!@#$%^&*()_+{}|:"<>?`1234567890-=[]\;',./AllCaps
