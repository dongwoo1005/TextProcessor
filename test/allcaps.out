A1
BC23
ABC123
D3SON
ALLCAPS~!@#$%^&*()_+{}|:"<>?`1234567890-=[]\;',./ALLCAPS
